src/sophys/common/utils/interfaces.py

```diff
@@ -0,0 +1,20 @@
+from abc import ABC, abstractmethod
+
+
+class IEnumValidator(ABC):
+    """Abstract base class for enum validation."""
+
+    @abstractmethod
+    def is_valid(self, enum_value) -> bool:
+        """Check if acquisition time is valid."""
+        pass
+
+    @abstractmethod
+    def format_to_enum(self, value) -> str:
+        """Format an input value to its enum correspondent"""
+        pass
+
+    @abstractmethod
+    def validate_and_format(self, value: float) -> bool:
+        """Validate and format acquisition time."""
```

Comment:
```suggestion
    def validate_and_format(self, value) -> str:
        """
        Validate and format the input value.
        
        Raises a 'ValueError' if validation fails.
        """
```

I'm actually not sure if this should be in the interface, since you just override it to combine `is_valid` and `validate_and_format` in the same way every time, and just modify the `ValueError` message in the subclass... annoying situation

Reply:
Yes, I think we can remove from this interface, if in the future we se applications we can simply add to it or create another more specific interface.

Reply:
Done at [this commit](https://github.com/cnpem/sophys-common/pull/36/commits/2cf653301197211caf3decb3088ca8b9f67bc155)